### PR TITLE
🌱 Set ClusterctlMoveHierarchyLabel on HelmChartProxy in CRD definition

### LINKED ISSUE #103: add support for "clusterctl move"
**User Story**

As a user I would like to be able to use "clusterctl move" to backup/restore/move cluster api configs to another place/cluster. To use caaph to replace ClusterResourceSet for network cni deployment. 

It seems "clusterctl move" currently works ok with ClusterResourceSet, but I don't see yaml being generated for hrp/hcp. 

**Detailed Description**

clusterctl move to also work with hrp/hcp. 

**Anything else you would like to add:**

none

/kind feature

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**: Have the HelmChartProxy controller set the ClusterctlMoveHierarchyLabel on HelmChartProxies so that it will be discovered by `clusterctl move`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #103

### CODE REVIEW

**Comment by Jont828** [comment]
@CecileRobertMichon @jackfrancis TL;DR HelmChartProxies need the `clusterctl.ClusterctlMoveHierarchyLabel` label attached in order for `clusterctl move` to work properly. Do you think it's appropriate to have the controller always add it, or instead leave it up to the user to opt-in?

**Comment by CecileRobertMichon** [comment]
@fabriziopandini do you have thoughts on this?

IMO if it can always be set so the user doesn't have to worry about it that's better, but what happens if the user is using the same HelmChartProxy across several clusters and moves 1 or some of the clusters?

note: if the goal is to set the label to every HelmChartProxy, the book indicates this can be done on the CRD itself so it applies to all HelmChartProxies instead of labelling each object.

> Note. clusterctl.cluster.x-k8s.io/move and clusterctl.cluster.x-k8s.io/move-hierarchy labels could be applied to single objects or at the CRD level (the label applies to all the objects).

https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html?highlight=move-hier#move

File: controllers/helmchartproxy/helmchartproxy_controller.go [hunk 1]
```diff
@@ -112,6 +108,11 @@ func (r *HelmChartProxyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	if helmChartProxy.Labels == nil {
+		helmChartProxy.Labels = make(map[string]string)
+	}
+	helmChartProxy.Labels[clusterctl.ClusterctlMoveHierarchyLabel] = "true"
```
[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply controllers/helmchartproxy/helmchartproxy_controller.go:114]: if we decide this approach is better we could also change the CAPZ AzureClusterIdentity approach

**Comment by fabriziopandini** [comment]
@fabriziopandini do you have thoughts on this?

I will suggest adding the label on the CRD definition, so it is done once and for all (no need to add on single CRs, reconcile it).

**Comment by Jont828** [comment]
How would we set a label on the CRD level? Would it be a kubebuilder tag?

**Comment by Jont828** [comment]
@fabriziopandini I tried adding the kubebuilder label based on the docs, but it doesn't seem to be working.  Are there any examples of this being used?
```
// +kubebuilder:metadata:labels=["clusterctl.cluster.x-k8s.io/move-hierarchy: true"]
```

**Comment by fabriziopandini** [comment]
> How would we set a label on the CRD level? Would it be a kubebuilder tag?

You can add it by creating a kustomize patch similar to [this one](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/main/config/crd/patches/cainjection_in_helmchartproxies.yaml).
Make sure that [name](  name: helmchartproxies.addons.cluster.x-k8s.io) is the name of the CRD you want to patch, replace [annotations](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/a31234e0db253dc158fd31739c1e8ad992ee49fe/config/crd/patches/cainjection_in_helmchartproxies.yaml#L5C1-L6) with labels: and the label you want to add.

Make sure to add your next patch to the [kustomization file](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/main/config/crd/kustomization.yaml)

**Comment by Jont828** [comment]
@fabriziopandini I've followed your suggestion and added the label to the CRD definition such that I have the following output. However, the label doesn't appear on the resources when I run `kubectl get` and they do not appear in the output of clusterctl move. Am I doing something wrong? 

```
$ kc describe crd helmchartproxies.addons.cluster.x-k8s.io
Name:         helmchartproxies.addons.cluster.x-k8s.io
Namespace:
Labels:       cluster.x-k8s.io/provider=helm
              clusterctl.cluster.x-k8s.io/move-hierarchy=true
Annotations:  cert-manager.io/inject-ca-from: caaph-system/caaph-serving-cert
              controller-gen.kubebuilder.io/version: v0.10.0
API Version:  apiextensions.k8s.io/v1
...
```

**Comment by Jont828** [comment]
@CecileRobertMichon @fabriziopandini I think I figured it out! The CRD label will only show up in the definition when it is installed through `make release-manifests` and not `make generate-manifests`, and it will only work with `clusterctl move` if the CRDs are installed through `clusterctl init --addon helm`.

Since that pulls from the GitHub releases, we will only know for sure that it works once this is merged and we cut a new CAAPH release. However, I tested that `addon-components.yaml` is produced to include the label, and that `clusterctl move` works once we manually edit the HelmChartProxy CRD definition to include the label, so I'm confident that this works and we can go ahead for review/merge!

File: config/crd/patches/clusterctl_move_label_in_helmchartproxies.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# The following patch adds a directive for certmanager to inject CA into the CRD
```
Comment (CecileRobertMichon) [inline config/crd/patches/clusterctl_move_label_in_helmchartproxies.yaml:1]: comment needs to be updated

**Comment by CecileRobertMichon** [comment]
one nit, otherwise lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**